File: telegram-bot-core/src/main/java/ru/tggc/telegrambotcore/access/checker/AccessChecker.java
package ru.tggc.telegrambotcore.access.checker;

import com.pengrad.telegrambot.model.Chat;
import com.pengrad.telegrambot.model.User;
import ru.tggc.telegrambotcore.dto.AccessResult;

import java.lang.reflect.Method;

public interface AccessChecker {

    AccessResult check(User from, Method method, Chat chat);
}

File: telegram-bot-core/src/main/kotlin/ru/tggc/telegrambotcore/access/checker/AccessChecker.kt
package ru.tggc.telegrambotcore.access.checker

import com.pengrad.telegrambot.model.Chat
import com.pengrad.telegrambot.model.User
import ru.tggc.telegrambotcore.dto.AccessResult
import java.lang.reflect.Method

interface AccessChecker {
    fun check(from: User, method: Method, chat: Chat): AccessResult
}
